.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,27 @@
+name: Python Tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  unit-tests:
+
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/project-codeflare/codeflare-sdk-precommit:v0.0.3
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install dependencies
```

Comment:
```yaml
      - name: Install dependencies
      - uses: actions/setup-python@v4
        with:
          python-version: '3.7'
          cache: 'poetry' # caching poetry dependencies
```

WDYT about caching dependencies here?

See: https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages

Reply:
@Fiona-Waters can you link the follow-up issue here?

Reply:
https://github.com/project-codeflare/codeflare-sdk/issues/345